mlflow/transformers.py

```diff
@@ -0,0 +1,919 @@
+    of the ``Pipeline`` class.
+    """
+    component_types = component_config.get(_COMPONENTS_BINARY_KEY, [])
+    for component_name in component_types:
+        component = getattr(pipeline, component_name)
+        component.save_pretrained(root_path.joinpath(component_name))
+    if processor:
+        processor.save_pretrained(root_path.joinpath(_PROCESSOR_KEY))
+
+
+def _load_component(root_path: pathlib.Path, component_key: str, component_type):
+    """
+    Loads an individual component object from local disk.
+    """
+    import transformers
+
+    components_dir = root_path.joinpath(_COMPONENTS_BINARY_KEY)
+    component_path = components_dir.joinpath(component_key)
+    component_instance = getattr(transformers, component_type)
+    return component_instance.from_pretrained(component_path)
+
+
+def _generate_base_flavor_configuration(
+    model,
+    task: str,
+) -> Dict[str, str]:
+    """
+    Generates the base flavor metadata needed for reconstructing a pipeline from saved
+    components. This is important because the ``Pipeline`` class does not have a loader
+    functionality. The serialization of a Pipeline saves the model, configurations, and
+    metadata for ``FeatureExtractor``s, ``Processor``s, and ``Tokenizer``s exclusively.
+    This function extracts key information from the submitted model object so that the precise
+    instance types can be loaded correctly.
+    """
+
+    _validate_transformers_task_type(task)
+
+    flavor_configuration = {
+        _TASK_KEY: task,
+        _INSTANCE_TYPE_KEY: _get_instance_type(model),
+        _MODEL_PATH_OR_NAME_KEY: _get_base_model_architecture(model),
+        _PIPELINE_MODEL_TYPE_KEY: _get_instance_type(model.model),
+    }
+
+    return flavor_configuration
+
+
+def _get_or_infer_task_type(model, task: Optional[str] = None) -> str:
+    """
+    Validates that a supplied task type is supported by the ``transformers`` library if supplied,
+    else, if not supplied, infers the appropriate task type based on the model type.
+    """
+    if task:
+        _validate_transformers_task_type(task)
+    else:
+        task = _infer_transformers_task_type(model)
+    return task
+
+
+def _infer_transformers_task_type(model) -> str:
+    """
+    Performs inference of the task type, used in generating a pipeline object based on the
+    underlying model's intended use case. This utility relies on the definitions within the
+    transformers pipeline construction utility functions.
+
+    :param model: Either the model or the Pipeline object that the task will be extracted or
+                  inferred from
+    :return: The task type string
+    """
+    from transformers import Pipeline
+    from transformers.pipelines import get_task
+
+    if isinstance(model, Pipeline):
+        return model.task
+    elif _isinstance_named_tuple(model):
+        try:
+            return get_task(model.model.name_or_path)
+        except RuntimeError as e:
```

Comment:
```suggestion
        except Exception as e:
```

Catching `Exception` might be safer in case huggingface suddenly changes `RunTimeError` to a different error.